tests/test_lane.py

```diff
@@ -827,230 +884,121 @@ def facetify(collection=Facets.COLLECTION_FULL,
-        facets = FeaturedFacets(minimum_featured_quality, True)
-
-        # This custom database query field will perform the calculation.
-        quality_field = facets.quality_tier_field().label("tier")
-
-        # Test it out by using it in a SELECT statement.
-        qu = self._db.query(
-            work_model, quality_field
-        ).join(
-            LicensePool,
-            LicensePool.id==work_model.license_pool_id
-        ).outerjoin(
-            CustomListEntry, CustomListEntry.work_id==work_model.works_id
-        )
-
-        expect_scores = {
-            # featured on list (11) + available (1)
-            awful_but_featured_on_a_list.sort_title: 12,
-
-            # featurable (5) + licensed (2) + available (1)
-            featurable.sort_title : 8,
-
-            # featurable (5) + licensed (2)
-            featurable_but_not_available.sort_title : 7,
-
-            # quality open access (2) + available (1)
-            decent_open_access.sort_title : 3,
-
-            # licensed (2) + available (1)
-            awful_but_licensed.sort_title : 3,
-
-            # available (1)
-            awful_open_access.sort_title : 1,
-        }
-
-        def best_score_dict(qu):
-            return dict((x.sort_title,y) for x, y in qu)
-
-        actual_scores = best_score_dict(qu)
-        eq_(expect_scores, actual_scores)
-
-        # If custom lists are not being considered, the "awful but
-        # featured on a list" work loses its cachet.
-        no_list_facets = FeaturedFacets(minimum_featured_quality, False)
-        quality_field = no_list_facets.quality_tier_field().label("tier")
-        no_list_qu = self._db.query(work_model, quality_field).join(
-            LicensePool,
-            LicensePool.id==work_model.license_pool_id
-        )
-
-        # 1 is the expected score for a work that has nothing going
-        # for it except for being available right now.
-        expect_scores[awful_but_featured_on_a_list.sort_title] = 1
-        actual_scores = best_score_dict(no_list_qu)
-        eq_(expect_scores, actual_scores)
-
-        # A low-quality work achieves the same low score if lists are
-        # considered but the work is not _featured_ on its list.
-        entry.featured = False
-        actual_scores = best_score_dict(qu)
-        eq_(expect_scores, actual_scores)
-
-
-    def test_apply(self):
-        """apply() orders a query randomly within quality tiers."""
-        high_quality_low_random = self._work(
-            title="High quality, low random", with_license_pool=True
-        )
-        high_quality_low_random.quality = 1
-        high_quality_low_random.random = 0
-
-        high_quality_high_random = self._work(
-            title="High quality, high random", with_license_pool=True
-        )
-        high_quality_high_random.quality = 0.7
-        high_quality_high_random.random = 1
+    def test_scoring_functions(self):        
```

Comment:
This wasn't previously tested in detail -- only the results of using it to generate grouped feeds.